server/src/test/java/org/elasticsearch/index/query/BoolQueryBuilderTests.java

```diff
@@ -277,14 +277,20 @@ public void testFromJson() throws IOException {
         assertEquals(query, "kimchy", ((TermQueryBuilder)queryBuilder.must().get(0)).value());
     }
 
+    public void testMinimumShouldMatchNumber() throws IOException {
+        String query = "{\"bool\" : {\"must\" : { \"term\" : { \"field\" : \"value\" } }, \"minimum_should_match\" : 1 } }";
+        BoolQueryBuilder builder = (BoolQueryBuilder) parseQuery(query);
+        assertEquals("1", builder.minimumShouldMatch());
+    }
+
     /**
      * test that unknown query names in the clauses throw an error
      */
     public void testUnknownQueryName() throws IOException {
         String query = "{\"bool\" : {\"must\" : { \"unknown_query\" : { } } } }";
 
-        ParsingException ex = expectThrows(ParsingException.class, () -> parseQuery(query));
-        assertEquals("unknown query [unknown_query]", ex.getMessage());
+        XContentParseException ex = expectThrows(XContentParseException.class, () -> parseQuery(query));
+        assertEquals("[1:41] [bool] failed to parse field [must]", ex.getMessage());
```

Comment:
Might be worth asserting on the "inner" exception too. Or something. Because this message isn't nearly a nice as the other one.

Reply:
++ I pushed f0f9b1542bd57f969d8c5149328b8ed368bebf05